prov/efa/src/rdm/efa_rdm_pkt_type.h

```diff
@@ -205,4 +205,34 @@ bool efa_rdm_pkt_type_contains_data(int pkt_type)
 	       efa_rdm_pkt_type_is_rta(pkt_type);
 }
 
+/**
+ * @brief determine whether a pack type is req packet
+ *
+ * @returns a boolean
+ */
+static inline
+bool efa_rdm_pkt_type_is_req(int pkt_type)
+{
+	if (pkt_type >= RXR_REQ_PKT_BEGIN) {
+		assert(pkt_type == RXR_EAGER_MSGRTM_PKT || pkt_type == RXR_EAGER_TAGRTM_PKT ||
+		       pkt_type == RXR_MEDIUM_MSGRTM_PKT || pkt_type == RXR_MEDIUM_TAGRTM_PKT ||
+		       pkt_type == RXR_LONGCTS_MSGRTM_PKT || pkt_type == RXR_LONGCTS_TAGRTM_PKT ||
+		       pkt_type == RXR_EAGER_RTW_PKT ||
+		       pkt_type == RXR_LONGCTS_RTW_PKT ||
+		       pkt_type == RXR_DC_EAGER_MSGRTM_PKT ||
+		       pkt_type == RXR_DC_EAGER_TAGRTM_PKT ||
+		       pkt_type == RXR_DC_MEDIUM_MSGRTM_PKT ||
+		       pkt_type == RXR_DC_MEDIUM_TAGRTM_PKT ||
+		       pkt_type == RXR_DC_LONGCTS_MSGRTM_PKT ||
+		       pkt_type == RXR_DC_LONGCTS_TAGRTM_PKT ||
+		       pkt_type == RXR_DC_EAGER_RTW_PKT ||
+		       pkt_type == RXR_DC_LONGCTS_RTW_PKT ||
+		       pkt_type == RXR_RUNTREAD_MSGRTM_PKT ||
+		       pkt_type == RXR_RUNTREAD_TAGRTM_PKT);
```

Comment:
Why do we need this here? `if (pkt_type >= RXR_REQ_PKT_BEGIN)` should cover it.

The assertion is more like a unit test IMO.

Reply:
I think this is what assertion is for, which is to fail if something should be happen actually happened.

Reply:
This assertion should be true in any case, not limited to this function branch.

My general comment is, invariants should be tested in dedicated unit tests rather than assertions in business logic.

Reply:
I am not sure. This is not a test.